### PR TITLE
Delayed gl init

### PR DESCRIPTION
**Pull request by dlyr**
draft PR to discuss issue #503

Closing #503

### CODE REVIEW

**Comment by nmellado** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ tests/ExampleApps/DrawPrimitivesApp/main.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/STORM/Radium-Engine/pullRequest?prid=5296592&bid=17337512)

**Comment by nmellado** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ tests/ExampleApps/DrawPrimitivesApp/main.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/STORM/Radium-Engine/pullRequest?prid=5296592&bid=17337512)

**Comment by nmellado** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ tests/ExampleApps/DrawPrimitivesApp/main.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/STORM/Radium-Engine/pullRequest?prid=5296592&bid=17337512)

**Comment by nmellado** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ tests/ExampleApps/DrawPrimitivesApp/main.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/STORM/Radium-Engine/pullRequest?prid=5296592&bid=17337512)

File: src/GuiBase/BaseApplication.cpp [hunk 1]
```diff
@@ -502,16 +529,16 @@ BaseApplication::~BaseApplication() {
     QDir().rmdir( m_exportFoldername.c_str() );
 }
 
-bool BaseApplication::loadPlugins( const std::string& pluginsPath,
+bool BaseApplication::loadPlugins( const std::string& pluginsPath2,
```
Comment (nmellado) [inline src/GuiBase/BaseApplication.cpp:532]: It is just a matter of style, but I don't like this name, may use `pluginPathCandidate` or equivalent ?

**Comment by nmellado** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ tests/ExampleApps/DrawPrimitivesApp/main.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/STORM/Radium-Engine/pullRequest?prid=5296592&bid=17337512)

File: src/GuiBase/BaseApplication.cpp [hunk 2]
```diff
@@ -223,13 +235,26 @@ BaseApplication::BaseApplication( int& argc,
...
+    // the rest of the initialization has to be done with valid opengl context,
+    // so defer after glInitialized.
+    QMetaObject::Connection* const connection = new QMetaObject::Connection;
+
+    *connection =
+        QObject::connect( m_viewer, &Ra::Gui::Viewer::glInitialized, [this, connection]() {
+            deferredInitialization();
+            QObject::disconnect( *connection );
+            delete connection;
+        } );
```
[the remark this answers is not in the document]
Comment (nmellado) [reply src/GuiBase/BaseApplication.cpp:252]: Up ?

File: src/GuiBase/BaseApplication.cpp (hunk 1 above)
Comment (dlyr) [reply src/GuiBase/BaseApplication.cpp:529]: sure this need polishing

File: src/GuiBase/BaseApplication.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (dlyr) [reply src/GuiBase/BaseApplication.cpp:252]: is dtor called if the application crashes ?

**Comment by nmellado** [comment]
What is the status of this PR and related issue ?

**Comment by nmellado** [comment]
@dlyr is this PR still relevant ?

**Pull request closed without merging**